packages/frontend-2/components/invite/dialog/Workspace.vue

```diff
@@ -0,0 +1,147 @@
+<template>
+  <div>
+    <InviteDialogSharedSelectUsers
+      v-model:open="isSelectUsersOpen"
+      title="Invite to Workspace"
+      :invites="invites"
+      :allowed-domains="allowedDomains"
+      invite-target="workspace"
+      @on-submit="onSelectUsersSubmit"
+      @on-cancel="isOpen = false"
+    >
+      <div v-if="showBillingInfo" class="text-body-2xs text-foreground-2 leading-5">
+        <p>
+          Inviting users may add seats to your current billing cycle. If there are
+          available seats, they will be used first. Your workspace is currently billed
+          for {{ memberSeatText }}{{ hasGuestSeats ? ` and ${guestSeatText}` : '' }}.
+        </p>
+      </div>
+    </InviteDialogSharedSelectUsers>
+  </div>
+</template>
+<script setup lang="ts">
+import { graphql } from '~/lib/common/generated/gql'
+import {
+  type InviteDialogWorkspace_WorkspaceFragment,
+  type WorkspaceInviteCreateInput,
+  type WorkspacePlans,
+  WorkspacePlanStatuses
+} from '~/lib/common/generated/gql/graphql'
+import type { InviteGenericItem } from '~~/lib/invites/helpers/types'
+import { emptyInviteGenericItem } from '~~/lib/invites/helpers/constants'
+import { Roles } from '@speckle/shared'
+import { useMixpanel } from '~/lib/core/composables/mp'
+import { mapMainRoleToGqlWorkspaceRole } from '~/lib/workspaces/helpers/roles'
+import { mapServerRoleToGqlServerRole } from '~/lib/common/helpers/roles'
+import { useInviteUserToWorkspace } from '~/lib/workspaces/composables/management'
+import { isPaidPlan } from '~/lib/billing/helpers/types'
+
+graphql(`
+  fragment InviteDialogWorkspace_Workspace on Workspace {
+    id
+    domainBasedMembershipProtectionEnabled
+    domains {
+      domain
+      id
+    }
+    plan {
+      status
+      name
+    }
+    subscription {
+      seats {
+        guest
+        plan
+      }
+    }
+  }
+`)
+
+const props = defineProps<{
+  workspace: InviteDialogWorkspace_WorkspaceFragment
+}>()
+const isOpen = defineModel<boolean>('open', { required: true })
+
+const mixpanel = useMixpanel()
+const inviteToWorkspace = useInviteUserToWorkspace()
+
+const isSelectUsersOpen = ref(false)
```

Comment:
This is not super relevant, in the branch there is only one submodal, but will make merging back later easier